GVRf/Framework/framework/src/main/jni/objects/textures/bitmap_transparency.cpp

```diff
@@ -0,0 +1,97 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/***************************************************************************
+ * JNI
+ ***************************************************************************/
+
+#include "bitmap_transparency.h"
+
+namespace gvr {
+
+bool bitmap_has_transparency(JNIEnv *env, jobject jbitmap) {
+    int result = -4;
+    bool transparency = false;
+	AndroidBitmapInfo info;
+    void *addrPtr = NULL;
+
+    result = AndroidBitmap_getInfo(env, jbitmap, &info);
+    if(result != ANDROID_BITMAP_RESUT_SUCCESS) {
+        LOGE("GVRBitmapTexture: unable to determine bitmap format in bitmap_transparency.cpp");
+        return false;
+    }
+
+    result = AndroidBitmap_lockPixels(env, jbitmap, &addrPtr);
+    if(result != ANDROID_BITMAP_RESUT_SUCCESS) {
+        LOGE("GVRBitmapTexture: unable to lock bitmap in bitmap_transparency.cpp");
+        return false;
+    }
+
+    uint32_t width = info.width;
+    uint32_t height = info.height;
+    uint32_t stride = info.stride;
+    int32_t format = info.format;
+    bool done = false;
+
+    if(format == ANDROID_BITMAP_FORMAT_A_8) {
+        const uint8_t *ptr = (uint8_t *)addrPtr;
+        while(height >= 0 && !done) {
+            if(*ptr < 255) {
+                transparency = true;
+                done = true;
+            }
+            ptr += stride;
+            height--;
+        }
+    } else if(format == ANDROID_BITMAP_FORMAT_RGBA_8888) {
+        const uint32_t *ptr = (uint32_t *)addrPtr; 
+        while(height >= 0 && !done) {
+            for(int x = 0; x < width && !done; x++) {
+                uint8_t alpha = ptr[x] << 24;
```

Comment:
The test with png_3_opaque always fails for me because the code determines that this particular png is non-opaque when it is supposed to opaque.

1. 0xFF808080 << 24 truncated to uint8_t yields a 0 for me; I think the result is implementation-defined. Shouldn't the shift be in the other direction actually?
2. When height becomes 0, ptr points to a 0 so even with 1. fixed, the png is marked as transparent. Are we getting out of bounds or is this png not quite as opaque as it should be?